kuma/users/tests/test_views.py

```diff
@@ -1295,26 +1297,203 @@ def test_account_refresh_token_saved_next_login(self):
         assert refresh_token == social_token.token_secret
 
 
-def test_missing_user_is_missing(db, client):
-    assert not User.objects.filter(username='missing').exists()
+def test_delete_user_login_always_required(db, client):
+    # Anonymous client gets redirected to sign in page.
     url = reverse('users.user_delete', kwargs={'username': 'missing'})
-    response = client.get(url)
+    response = client.get(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert reverse('account_login') in response['Location']
+
+
+def test_delete_user_not_allowed(db, user_client, wiki_user_2):
+    # A username that doesn't exist.
+    url = reverse('users.user_delete', kwargs={'username': 'missing'})
+    response = user_client.get(url, HTTP_HOST=settings.WIKI_HOST)
     assert response.status_code == 404
-    assert_no_cache_header(response)
 
+    # Attempting to delete someone else.
+    url = reverse('users.user_delete', kwargs={'username': wiki_user_2.username})
+    response = user_client.get(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 403
 
-@pytest.mark.parametrize('user_case', ['wrong_user', 'right_user'])
-def test_user_can_delete(wiki_user, wiki_user_2, user_client, user_case):
-    if user_case == 'wrong_user':
-        user = wiki_user_2
-        expected_status = 403
-    else:
-        user = wiki_user
-        expected_status = 200
-    url = reverse('users.user_delete', kwargs={'username': user.username})
-    response = user_client.get(url)
-    assert response.status_code == expected_status
-    assert_no_cache_header(response)
+
+def test_delete_user_with_no_revisions(db, user_client, wiki_user):
+    # sanity check fixtures
+    assert not Revision.objects.filter(creator=wiki_user).exists()
+    assert not AttachmentRevision.objects.filter(creator=wiki_user).exists()
+    url = reverse('users.user_delete', kwargs={'username': wiki_user.username})
+    response = user_client.get(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 200
+    response = user_client.post(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert not User.objects.filter(username=wiki_user.username).exists()
+
+
+def test_delete_user_donate_attributions(db, user_client, wiki_user):
+
+    # Pretend the user logged in with GitHub
+    SocialAccount.objects.create(
+        user=wiki_user,
+        provider='github',
+        extra_data=dict(
+            email=wiki_user.email,
+            avatar_url='https://avatars0.githubusercontent.com/yada/yada',
+            html_url="https://github.com/{}".format(wiki_user.username)
+        )
+    )
```

Comment:
You can use the `wiki_user_github_account` pytest fixture instead of creating this yourself.